File: py_ver/monochromatic_example.py
#!/usr/bin/env python3

import numpy as np
from numpy import linalg as la
from monochromatic import monochromatic_approx

if __name__ == "__main__":
    W = np.abs(np.random.rand(32, 3, 3, 3))
    print("||W|| == " + str(la.norm(W)))

    [Wapprox, Wmono, colors, perm] = monochromatic_approx(W, num_colors=4, even=False)
    L2_err = la.norm(W - Wapprox) / la.norm(W)

    print("||W - Wapprox|| / ||W|| == " + str(L2_err))

File: compress_conv/monochromatic_example.m
W = abs(randn(96, 7, 7, 3));  % make random weights with 3 input color channels, 96 output channelsand 7x7 filters

% Compute approximation
fprintf('||W|| = %f \n', norm(W(:)));
args.num_colors = 6;
args.even = false; % default: true
[Wapprox, Wmono, colors, perm] = monochromatic_approx(double(W), args);
L2_err = norm(W(:) - Wapprox(:)) / norm(W(:));
fprintf('||W - Wapprox|| / ||W|| = %f (*note* error should be high since weights are random and thus have no structure...\n', L2_err);

